src/modules/dashboard/components/ExpenditurePage/ExpenditureSettings/ExpenditureStreamingForm.tsx

```diff
@@ -39,14 +40,16 @@ export const MSG = defineMessages({
   },
 });
 
-const endDateTypes = [
+const endDateOptions: DatePickerOption[] = [
   {
     label: MSG.whenCancelled,
     value: ExpenditureEndDateTypes.WhenCancelled,
+    hideDatePicker: true,
   },
   {
     label: MSG.limitIsReached,
     value: ExpenditureEndDateTypes.LimitIsReached,
+    hideDatePicker: true,
   },
   {
     label: MSG.fixedTime,
```

Comment:
Add `hideDatePicker: false` to this one for consistency? And then remove as optional property from `DatePickerOption`? Purely a matter of personal preference.